src/browser.ts

```diff
@@ -0,0 +1,170 @@
+import { ChildProcess, spawn } from "child_process";
+import { EventEmitter } from "events";
+import * as mktemp from "mktemp";
+import * as path from "path";
+import * as os from "os";
+import * as rimraf from "rimraf";
+import * as fs from "fs";
+import * as connectToChrome from "chrome-remote-interface";
+
+import {
+  default as DefaultBrowserResolver,
+  BrowserResolverOptions,
+  BrowserResolver,
+  BrowserResolution
+} from "./browser_resolver";
+
+export interface BrowserOptions {
+  command?: string;
+  chromiumSrcDir?: string;
+  createResolver?: (options: BrowserResolverOptions) => BrowserResolver;
+}
+
+function createDefaultResolver(options: BrowserResolverOptions) {
+  return new DefaultBrowserResolver(options);
+}
+
+// Phases
+// closed
+// starting
+// started and waiting for port
+// started and connected
+
+export enum BrowserState {
+  CLOSED,
+  WAITING_FOR_PORT,
+  CONNECTING,
+  CONNECTED,
+}
+
+export default class Browser extends EventEmitter {
+  state: BrowserState = BrowserState.CLOSED;
+  command: string;
+  isContentShell: boolean;
+  dataDir: string;
+  process: ChildProcess;
+  activePortFile: string;
+  port: number;
+  remoteInterface: any;
+
+  constructor(browserType: string, options?: BrowserOptions) {
```

Comment:
I don't think node supports that syntax, and last time I tried typescript didn't transpile it.